### PR TITLE
Fix clear button on `Array[Node]`.

### LINKED ISSUE #96273: Exported Vars: Cannot clear item from array when hint is Array[Node]
### Tested versions

v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - Windows 10.0.22631 - Vulkan (Forward+)

### Issue description

When exporting a variable with the hint `Array[Node]` and adding an item to it, then on that item clicking the 3 dots then clicking `Clear`, this will show the following error:

```
 Attempted to set a variable of type 'NodePath' into a TypedArray of type 'Object'.
  core/variant/array.cpp:429 - Condition "!_p->typed.validate(value, "set")" is true.
```

### Steps to reproduce

1. Create a new project (v4.3.stable is the only one I tested)
2. Add a node
3. Add a script
4. Write `@export var test : Array[Node]`
5. In the inspector, add an item to the array (select any node)
6. Now clear that item only using `3 Dots` > `Clear`
7. The error above should display in `Output`

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by ajreckof**
Fix #96273

### CODE REVIEW

File: editor/editor_properties.cpp [hunk 1]
```diff
@@ -2703,7 +2703,7 @@ void EditorPropertyNodePath::_update_menu() {
 void EditorPropertyNodePath::_menu_option(int p_idx) {
 	switch (p_idx) {
 		case ACTION_CLEAR: {
-			emit_changed(get_edited_property(), NodePath());
+			emit_changed(get_edited_property(), Variant());
```
Comment (kleonc) [inline editor/editor_properties.cpp:2706]: `Variant()` breaks clearing for `@export var array: Array[NodePath]`.
```suggestion
			if (editing_node) {
				emit_changed(get_edited_property(), Variant());
			} else {
				emit_changed(get_edited_property(), NodePath());
			}
```

Comment (ajreckof) [reply editor/editor_properties.cpp:2706]: Thanks for catching this. I thought there was an implicit conversion from nil to the null NodePath.

**Review by kleonc — APPROVED** [review]
Fixes the issue.

**Review by dalexeev — APPROVED** [review]
Looks right to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**